graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/FederationDirectives.java

```diff
@@ -26,82 +26,136 @@
-            .type(new NonNullType(new TypeName(_FieldSet.typeName)))
-            .build();
+    /* Directive locations */
 
     private static final DirectiveLocation DL_OBJECT = newDirectiveLocation()
             .name("OBJECT")
             .build();
+
     private static final DirectiveLocation DL_INTERFACE = newDirectiveLocation()
             .name("INTERFACE")
             .build();
+
     private static final DirectiveLocation DL_FIELD_DEFINITION = newDirectiveLocation()
             .name("FIELD_DEFINITION")
             .build();
 
+    /* fields: _FieldSet */
+
+    private static final GraphQLArgument fieldsArgument = newArgument()
+            .name("fields")
+            .type(new GraphQLNonNull(_FieldSet.type))
+            .build();
+
+    private static final GraphQLArgument fieldsArgument(String value) {
+        return newArgument(fieldsArgument)
+                .value(value)
+                .build();
+    }
+
+    private static final InputValueDefinition fieldsDefinition = newInputValueDefinition()
+            .name("fields")
+            .type(new NonNullType(new TypeName(_FieldSet.typeName)))
+            .build();
+
+    /* directive @key(fields: _FieldSet!) on OBJECT | INTERFACE */
+
     public static final String keyName = "key";
+
     public static final GraphQLDirective key = newDirective()
             .name(keyName)
             .validLocations(OBJECT, INTERFACE)
             .argument(fieldsArgument)
             .build();
+
+    public static final GraphQLDirective key(String fields) {
+        return newDirective(key)
+                .argument(fieldsArgument(fields))
+                .build();
+    }
+
     public static final DirectiveDefinition keyDefinition = newDirectiveDefinition()
             .name(keyName)
             .directiveLocations(Arrays.asList(DL_OBJECT, DL_INTERFACE))
             .inputValueDefinition(fieldsDefinition)
             .build();
 
+    /* directive @external on FIELD_DEFINITION */
+
     public static final String externalName = "external";
+
     public static final GraphQLDirective external = newDirective()
             .name(externalName)
-            .validLocations(OBJECT, FIELD_DEFINITION)
+            .validLocations(FIELD_DEFINITION)
             .build();
+
     public static final DirectiveDefinition externalDefinition = newDirectiveDefinition()
             .name(externalName)
-            .directiveLocations(Arrays.asList(DL_OBJECT, DL_FIELD_DEFINITION))
+            .directiveLocations(Arrays.asList(DL_FIELD_DEFINITION))
             .build();
 
+    /* directive @requires(fields: _FieldSet!) on FIELD_DEFINITION */
+
     public static final String requiresName = "requires";
+
     public static final GraphQLDirective requires = newDirective()
             .name(requiresName)
             .validLocations(FIELD_DEFINITION)
             .argument(fieldsArgument)
             .build();
+
+    public static final GraphQLDirective requires(String fields) {
```

Comment:
usually public APIs have some kind of javadoc, right? We probably don't need it to explain the spec, but we could add a note to usage and maybe link to the spec doc?